iphone/Maps/UI/Search/Filters/MWMSearchHotelsFilterViewController.mm

```diff
@@ -502,13 +513,14 @@ - (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPa
   }
   [Statistics logEvent:kStatSearchFilterClick
         withParameters:@{kStatCategory: kStatHotel, kStatType: typeString}];
-  m_selectedTypes.emplace_back(type);
+  m_selectedTypes.emplace(type);
 }
 
 - (void)collectionView:(UICollectionView *)collectionView didDeselectItemAtIndexPath:(NSIndexPath *)indexPath
 {
   auto const type = kTypes[indexPath.row];
-  m_selectedTypes.erase(remove(m_selectedTypes.begin(), m_selectedTypes.end(), type));
+  if (m_selectedTypes.find(type) != m_selectedTypes.end())
+    m_selectedTypes.erase(type);
```

Comment:
можно не проверять что элемент есть перед удалением:
```C++
std::unordered_set::erase

size_type erase( const key_type& key ); | (3) | (since C++11)

3) Removes the element (if one exists) with the key equivalent to key.
```
если сначала поискать, а потом удалить получается лишний поиск